fractal/cmd/_task.py

```diff
@@ -143,12 +144,12 @@ async def patch_task(
     meta_file: Optional[str] = None,
 ) -> BaseInterface:
 
-    try:
-        task_id = int(task_id_or_name)
-        if version:
-            logging.warning("Task Version is ignored because Task ID provided")
-    except ValueError:
-        task_name = task_id_or_name
+    if task_id and version:
+        logging.warning(
+            "Too many arguments: cannot provide both `task_id` and `version`."
+        )
+        sys.exit(1)
+    elif task_name:
```

Comment:
In principle this `elif` could be an `else`, because one of the two is required.
If you write it as `elif`, then there should also be an `else`

Reply:
I put `elif` because the first if checks for `task_id and version`. But we can have `task_id`, `task_name` and no `version`

Reply:
My bad, this makes sense. I just don't find it very readable, plus I'm a bit confused by the if/elif (without a final else).

What about the following?
```python
if task_id:
    if task_version:
        logging..
        sys.exit(1)
else:
    task_id = ...
do things with task_id
```

I just find a bit more readable.

If you prefer to stick with the if/elif (without else), then I would at least add some clear comments.